modules/cloudiot/README.md

```diff
@@ -0,0 +1,86 @@
+# terraform-google-pubsub for Cloud IoT
+
+## Overview
+
+In the early stage, this module will simply be implemented by using [existing pubsub module](https://github.com/terraform-google-modules/terraform-google-pubsub) and [one resource](https://www.terraform.io/docs/providers/google/r/cloudiot_registry.html).
+
+This module may not improve drastically efficiency. However, after implementing new resources like [terraform-provider-google#1495](https://github.com/terraform-providers/terraform-provider-google/issues/1495), this module will get more efficient by adopting the new resources.
+
+## Usage
+
+Let's seeing a simple usage of the module. See also a simple setup provided in the example directory.
+
+```hcl
+module "iot" {
+  source     = "terraform-google-modules/pubsub/iot"
+  name       = "sample-iot"
+  region     = "us-central1"
+  project_id = "tf-project"
+
+  mqtt_enabled_state = "MQTT_ENABLED"
+  http_enabled_state = "HTTP_DISABLED"
+  event_notification_config = {
+    topic = "iot-event-topic"
+    pull_subscriptions = [
+      {
+        name = "iot-event-pull"
+        ack_deadline_seconds = 20
+      },
+    ]
+  }
+  state_notification_config = {
+    topic = "iot-state-topic"
+    pull_subscriptions = [
+      {
+        name = "iot-state-pull"
+        ack_deadline_seconds = 20
+      },
+    ]
+  }
+}
+```
+
+[^]: (autogen_docs_start)## Inputs
```

Comment:
this seems to break markdown rendering so I'll fix it soon

Reply:
done: https://github.com/terraform-google-modules/terraform-google-pubsub/commit/71a957b452f7df6a9f0d1417853bd0de414a8ec7